### PR TITLE
 WZ-37394: move azure-creds secret part 1

### PR DESCRIPTION
**Pull request by eyal-moscovici**

### CODE REVIEW

File: git-proxy/values.yaml [hunk 1]
```diff
@@ -21,6 +21,7 @@ blob:
   #    environment:
   #    useInstanceMetadata:
   #    useFederatedToken:
+  azureCredsSecretName:
```
Comment (yarinm) [inline git-proxy/values.yaml:24]: default value should be the secret name we are already creating in this chart? (and use the same key for the secret being created)

**Review by yarinm — CHANGES_REQUESTED** [review]

File: git-proxy/values.yaml (hunk 1 above)
Comment (eyal-moscovici) [reply git-proxy/values.yaml:24]: I can rename, but azureCredsSecretName == nil means don't use azure creds secret created in wiz-outpost-config. 
I made it look and function like outpost-csi-files.

Comment (eyal-moscovici) [reply git-proxy/values.yaml:24]: I will add a createAzureCredsSercret flag

Comment (eyal-moscovici) [reply git-proxy/values.yaml:24]: thought about it and adding createAzureCredsSercret or setting the secret name breaks backwards compatibility

File: wiz-outpost-configuration/templates/azureWorkerCredentials.secret.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+{{- if .Values.azureWorkerCredentials }}
+{{/* publish the secret for git-proxy / external secrets */}}
+apiVersion: v1
+kind: Secret
+type: Opaque
+metadata:
+  name: azure-creds
```
Comment (yarinm) [inline wiz-outpost-configuration/templates/azureWorkerCredentials.secret.yaml:7]:
```suggestion
  name: {{ .Values.blob.azureCredsSecretName }} 
```

**Review by yarinm — CHANGES_REQUESTED** [review]

**Review by yarinm — APPROVED** [review]

**Pull request merged**